specification.md

```diff
@@ -0,0 +1,89 @@
+# OpenTelemetry/Prometheus Compatibility Specification
+
+Status: [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/document-status.md)
+
+## Abstract
+
+OpenTelemetry is aiming to provide compatibility with
+Prometheus and OpenMetrics. This document explains the
+extent of the support. OpenTelemetry collector and libraries
+will align with the compatibility requirements defined in
+this spec.
+
+## Goals
+
+* OpenTelemetry collector can be used as a drop-in replacement
+  for Prometheus server to scrape and export metrics data.
+* OpenTelemetry collector should export OLTP-compatible Prometheus
+  time series to OpenTelemetry metrics exporters.
+* OpenTelemetry libraries should implement exporters to publish
+  Prometheus metrics.
+
+## Differences and Limitations
+
+OpenTelemetry and Prometheus/OpenMetrics have different design
+goals and this reflects in the data model and implementation
+details. This section summarizes a few key differences.
+
+* **Pull vs push**: Prometheus is mainly designed for pull
+  whereas OpenTelemetry primarily is designed for push. This
+  difference causes how the state is maintained throughout the
+  collection including how it’s handled in the OpenTelemetry
+  collector.
+* **Cumulative vs delta**: OpenTelemetry supports delta temporality
+  whereas Prometheus always expects absolute/cumulative values. This
+  breaks some components that produce and communicate deltas where
+  cumulatives cannot be rebuilt before being exported to Prometheus.
+* **Histogram boundaries**: Prometheus histogram boundaries are by
+  lower equal (le) while OpenTelemetry histogram boundaries are
+  greater equal (ge). (This difference will be resolved via #18)
+* **Semantic conventions**: OpenTelemetry predefines semantic
+  conventions to collect additional metadata with telemetry data.
+  Prometheus users don’t follow the same conventions and Prometheus
+  client library provided data may lack the semantic conventions
+  available in OpenTelemetry libraries.
+
+## Compatibility Requirements
+
+Given the number of fundamental design goals between OpenTelemetry
+and Prometheus, our aim is to close the gaps where possible and
+make the right compromises to meet the goals defined in this document.
+OpenTelemery and Prometheus won’t be fully compatible but we will
```

Comment:
Can we be explicit about what is meant here?

Reply:
I do not believe we should be making any compromises when we say that we support Prometheus. 

The proposal being built up here, is that an OpenTelemetry collector can eventually be configured with a number of pipeline stages, using receivers, processors, and exporters, to behave **exactly** as a Prometheus server would in addition to all of the other things it's capable of doing.

I understand there has been a mixed message.

For example, I know that some members of the OTel group have suggested we not support OpenMetrics export, which would mean not supporting the Prometheus federation design. We should be saying that we can do this, simply put, as it's not much different than supporting a Prometheus exporter inside the SDKs.

Reply:
NB: Federation implies that data went through a Prometheus and received cleanups, sanity checking, etc. That will need an in-depth look on its own to make sure we don't accidentally get something wrong.

Reply:
Our goal in working on specs like this is that we can replace "went through a Prometheus and received cleanups, sanity checking, etc" with a firm statement about what is required. I like how @rakyll wrote

> To the OpenTelemetry collector, an OpenTelemetry target and a Prometheus instrumented target is indistinguishable.
  
I'd add something high-level like, "To the Prometheus server, an OpenTelemetry collector and another Prometheus server are indistinguishable for establishing a federation."